test/final.test.js

```diff
@@ -8,6 +8,35 @@ const { sleep, getPathToNull } = require('./helper')
 // will be emitted. Let's raise this so we do not scare everybody.
 process.setMaxListeners(100)
 
+test('should show warning for pino.final on node 14+', async ({ equal }) => {
+  const dest = pino.destination({ dest: getPathToNull(), sync: false })
+  dest.flushSync = () => {}
+  const instance = pino(dest)
+
+  pino.final(instance, (_, finalLogger) => {
+    finalLogger.info('hello')
+  })()
+
+  const major = Number(process.versions.node.split('.')[0])
+  let isWarningEmitted = false
+
+  function onWarning (warning) {
+    isWarningEmitted = true
+    equal(warning.code, 'PINODEP009')
+  }
+
+  if (major >= 14) process.on('warning', onWarning)
+
+  instance.info('hello')
+  await sleep(10)
```

Comment:
This seems fragile. Instead of using an `async` test, how about using the callback form and issuing `t.end()` when the test has been completed?

Reply:
Makes sense. I have changed the code to call `t.end()` in the warning callback instead of using sleep.